src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -799,6 +799,31 @@ def basic_open_iso (f : R) : (structure_sheaf R).presheaf.obj (op (basic_open f)
   CommRing.of (localization.away f) :=
 (as_iso (show CommRing.of _ ⟶ _, from to_basic_open R f)).symm
 
+@[elementwise] lemma to_global_factors : to_open R ⊤ =
+  (CommRing.of_hom (algebra_map R (localization.away (1 : R)))) ≫ (to_basic_open R (1 : R)) ≫
+  (structure_sheaf R).presheaf.map (eq_to_hom (basic_open_one.symm)).op :=
+begin
+  change to_open R ⊤ = (to_basic_open R 1).comp _ ≫ _,
+  unfold CommRing.of_hom,
+  rw [localization_to_basic_open R, to_open_res],
+end
```

Comment:
I'm not thrilled if at the first introduction of `CommRing.of_hom` we're having to `unfold` it by hand. However I'm not immediately seeing something better here. Thoughts, @jcommelin?

Reply:
I could remove that line and change the next to `erw`, but I'm not sure if that's better.

Reply:
Hmmz, I think there is an API barrier between "uncategorical" homs and categorical homs. So when we cross it, this leaves a little trace in the proof.
I think that until now, mathlib has been somewhat unclear about this API barrier, but that has also been causing some nuisances.

Now that I think of it: should `CommRing.of_hom` and friends be defined in terms of `bundled.of_hom` (to be defined). That might unify their (small) APIs.

Reply:
I don't really see how to make a unified definition. None of these functions do anything: they are all just the identity, but with different type signatures to help type-checking.

Reply:
Certainly `unfold` is better than `erw`. :-)